dlt/destinations/athena/configuration.py

```diff
@@ -0,0 +1,14 @@
+from typing import Final, Optional
+from dlt.common.typing import TSecretStrValue
+
+from dlt.common.configuration import configspec
+from dlt.common.destination.reference import DestinationClientDwhConfiguration
+from dlt.common.configuration.specs import  AwsCredentials
```

Comment:
I believe I'm mistaken here and just lack a full understanding.

After some testing however. If we create a boto3 session without specifying the profile name. The profile name comes out to be default. I.e.

```
>>> session = boto3.Session(region_name = 'us-west-2')
>>> session.profile_name
'default'
```

OR

```
>>> from pyathena import connect
>>> athena_client = connect(work_group="dev-dlt-athena-workgroup", region_name="us-west-2")
>>> athena_client.session
Session(region_name='us-west-2')
>>> athena_client.session.profile_name
'default'
```

AND from the a pod on our cluster the profile shows to be 'default' as well. The issue is that if there is no shared credentials file or config file but a profile name is provided then boto3 will not succeed in finding the profile. In our case there isn't one.

My suggestion is that we don't set the profile_name from the session and make it an intentional configuration.

Reply:
See here: 
https://github.com/lukas-gust/dlt/blob/7fd0b44170dc00394dfd6dc7301fbf83f3e4bac6/dlt/common/configuration/specs/aws_credentials.py#L53

Reply:
I'm OK with doing that just let's make sure that all is clear. here's the part that is responsible for setting up the profile name (and other params) from boto3 session
```
def on_partial(self) -> None:
        # Try get default credentials
        session = self._to_session()
        if self._from_session(session) and not self.is_partial():
            self.resolve()
```
so first it passes the parameters provided by user to  instantiate new boto3 Session. if user sets the profile it will be passed to the session. If such profile is not found then boto3 raises:
```
botocore.exceptions.ProfileNotFound: The config profile (fake_profile) could not be found
```
as in the test below:
```
    environment["CREDENTIALS__PROFILE_NAME"] = "fake_profile"
    with pytest.raises(botocore.exceptions.ProfileNotFound):
        resolve_configuration(AwsCredentials())
```
so your problem is that boto3 returns incorrect profile name? so when we actually try to create new session later it will fail?

Reply:
The problem seems to be if you don't give it a profile name it will set it to "default" and subsequently pass that argument to pyathena. That shouldn't happen in the cases where someone needs to go beyond config or credential files like I do. If I don't give a profile name then it should be set to `None` not `"default"`.

Reply:
OK get it. @sh-rp I'll do a separate PR with it. it is just one line but will make several tests to fail which I'd fix separately